src/vcpkg/binarycaching.cpp

```diff
@@ -951,14 +951,22 @@ namespace
         static constexpr StringLiteral m_accept_header = "Accept: application/json;api-version=6.0-preview.1";
     };
 
+    enum class CacheResult
+    {
+        hit,
+        miss,
+    };
+
+    static CacheResult unit_to_cache_hit(const Unit&) { return CacheResult::hit; }
+
     struct IObjectStorageTool
     {
         virtual ~IObjectStorageTool() = default;
 
         virtual LocalizedString restored_message(size_t count,
                                                  std::chrono::high_resolution_clock::duration elapsed) const = 0;
-        virtual ExpectedL<Unit> stat(StringView url) const = 0;
-        virtual ExpectedL<Unit> download_file(StringView object, const Path& archive) const = 0;
+        virtual ExpectedL<CacheResult> stat(StringView url) const = 0;
```

Comment:
Should `stat` return `CacheAvailability` instead?

Reply:
Do you mean `ExpectedL<CacheAvailability>`, or plain `CacheAvailability`? In either case, how should we treat `unknown` in the call sites?

Reply:
OK, I've found your suggestion for `precheck()`, but it's still unclear to me how to handle it in `AwsStorageTool::download_file()`.  Should I simply return `RestoreResult:: unavailable`?

Reply:
> Do you mean `ExpectedL<CacheAvailability>`, or plain `CacheAvailability`? In either case, how should we treat `unknown` in the call sites?

Sorry, I meant `ExpectedL<CacheAvailability>` :)